OVERVIEW.md

```diff
@@ -0,0 +1,66 @@
+# Overview #
+
+Here it is described a brief overview of xdist's internal architecture.
+
+
+`xdist` works by spawning one or more **worker nodes**, which are controlled
```

Comment:
Should we drop "node" from the terminology? A "node" implies a separate host, which is not always the case. How about just a "master" and "workers"? The workers can just be separate processes on the same host, or they can be remote hosts, as you explain later.

(I think @hpk42 was advocating for this change in another issue.)